core-bundle/src/Security/Authentication/Token/TokenChecker.php

```diff
@@ -97,14 +101,36 @@ public function getBackendUsername(): ?string
         return $token->getUser()->getUserIdentifier();
     }
 
+    /**
+     * Tells whether the front end preview can be accessed.
+     */
+    public function isPreviewAllowed(): bool
+    {
+        if ($this->hasBackendUser()) {
+            return true;
+        }
+
+        $token = $this->getToken(self::FRONTEND_FIREWALL);
+
+        if (!$token instanceof FrontendPreviewToken) {
+            return false;
+        }
+
+        if (null === $token->getPreviewLinkId()) {
```

Comment:
Any reason for this? If preview is never working without that, why is it not a mandatory argument in the `FrontendPreviewToken`?

Reply:
`$token->getPreviewLinkId()` is null for backend users and for existing sessions of previous Contao versions.